tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,974 @@
+                # Commit transaction or abort it
+                topic.producer.commit_transaction()
+                topic.index += 1
+                sent_count += 1
+                if topic.index % CONSUMER_LOGGING_THRESHOLD == 0:
+                    logger.debug(f"..sent {topic.index} to {topic.name}")
+            except ck.KafkaException:
+                # In case of any exception, abort it
+
+                # TODO: handle retry message logic
+
+                topic.producer.abort_transaction()
+                logger.warning(f"Transaction {topic.transaction_id} aborted")
+
+            # save time for this message
+            topic.last_message_ts = datetime.now().timestamp()
+
+            # exit if terminate flag is set
+            if topic.terminate:
+                logger.warning("Got terminate signal. Exiting")
+                break
+        # Return stats
+        return sent_count
+
+    @staticmethod
+    def _worker_thread(func, workers: int, topics: dict, total_messages: int,
+                       logger: logging.Logger):
+        pool = ThreadPoolExecutor(workers, "stream_worker")
+        msgs_processed = 0
+        # Sending loop
+        while msgs_processed < total_messages:
+            # Get processed message count from worker threads
+            for msg_count in pool.map(func, repeat(logger),
+                                      list(topics.values())):
+                msgs_processed += msg_count
+            # Log pretty name of underlying func
+            logger.info(f"{func.__qualname__}, "
+                        f"processed so far {msgs_processed}")
+            # Check termination flag in all topics before the next chunk
+            if any([t.terminate for t in topics.values()]):
+                logger.warning("Got terminate signal, "
+                               "exiting from message processing")
+        logger.info("End of processing messages.")
+
+    def produce(self, wait=True):
+        """Starts produce messages thread
+
+        Args:
+            wait (bool, optional): Wait for produce thread to finish or not.
+            Defaults to True.
+        """
+        self.logger.info("Start of sending messages")
+        thread = threading.Thread(name="stream_produce_thread",
+                                  target=self._worker_thread,
+                                  args=(self._send_transaction, self.workers,
+                                        self.topics, self.total_messages,
+                                        self.logger))
+        thread.start()
+        self.produce_thread = thread
+        if wait:
+            thread.join()
+        return
+
+    def init_consumers(self):
+        """Precreates topic status lists for Cosuming thread.
+        One Consumer per worker thread.
+        """
+        self.logger.info("Initializing consumers")
+        for name in self.topics_config.names:
+            topic_config = {
```

Comment:
i'd rename this from topic_config.. it is confusing because topic_config is used with a different meaning in RP code.